plugins/caddy/httpcache_test.go

```diff
@@ -114,6 +114,39 @@ func TestQueryString(t *testing.T) {
 	}
 }
 
+func TestCacheKeys(t *testing.T) {
+	tester := caddytest.NewTester(t)
+	tester.InitServer(`
+	{
+		admin localhost:2999
+		order cache before rewrite
+		http_port     9080
+		https_port    9443
+		cache 
+	}
+	localhost:9080 {
+		route /query-string {
+			cache {
+				cache_keys {
+					query= {
+						disable_query
+					}
+				}
+			}
+			respond "Hello, query string!"
+		}
+	}`, "caddyfile")
+
+	resp1, _ := tester.AssertGetResponse(`http://localhost:9080/query-string?query=string`, 200, "Hello, query string!")
+	if resp1.Header.Get("Cache-Status") != "Souin; fwd=uri-miss; stored; key=GET-http-localhost:9080-/query-string" {
+		t.Errorf("unexpected Cache-Status header %v", resp1.Header)
+	}
+	resp2, _ := tester.AssertGetResponse(`http://localhost:9080/query-string?foo=bar`, 200, "Hello, query string!")
+	if resp2.Header.Get("Cache-Status") != "Souin; fwd=uri-miss; stored; key=GET-http-localhost:9080-/query-string?foo=bar" {
+		t.Errorf("unexpected Cache-Status header %v", resp2.Header)
+	}
```

Comment:
If you disable query within the cache_key then you can expect two stored responses because it should generate a key like that `GET-http-localhost:9080-/query-string` for both so the first one will receive a response that is stored and the second one will get a cached response.

Reply:
Thanks for the quick reply.
If I run the new test case alone with `go test -run TestCacheKeys`, then it passes, but without the `run` filter the same test fails. Could be a persistent state between tests?

I expected the new test case to be valid because the `query=` regexp matches the first request and modifies the key generation to exclude the query string, and the key becomes `GET-http-localhost:9080-/query-string`. The second request does not match the regexp so the key generation should include the query string and the key becomes `GET-http-localhost:9080-/query-string?foo=bar`. Since it is a different key its a miss and store.
At least this was my understanding of the docs. What is the expected behavior in this case?